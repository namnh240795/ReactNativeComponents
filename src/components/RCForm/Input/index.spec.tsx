import { fireEvent, render, screen } from '@testing-library/react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import InputText from '.';

test('Render empty input', () => {
  const tree = renderer.create(<InputText />);
  expect(tree).toMatchSnapshot();
});

test('Render padding all input', () => {
  const tree = renderer.create(<InputText padding={8} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render padding hv input', () => {
  const tree = renderer.create(<InputText padding={[0, 2]} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render padding width array input', () => {
  const tree = renderer.create(<InputText padding={[0, 2, 0, 4]} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render margin all input', () => {
  const tree = renderer.create(<InputText margin={8} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render margin hv input', () => {
  const tree = renderer.create(<InputText margin={[0, 2]} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render margin width array input', () => {
  const tree = renderer.create(<InputText margin={[0, 2, 0, 4]} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render disable input', () => {
  const tree = renderer.create(<InputText disabled />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render input with suffix', () => {
  const tree = renderer.create(<InputText suffix={<>suffix</>} />);
  expect(tree).toMatchSnapshot();
});

test('Render input with prefix', () => {
  const tree = renderer.create(<InputText prefix={<>paste</>} />);
  expect(tree).toMatchSnapshot();
});

test('change text input event', () => {
  const onChangeMock = jest.fn();
  render(<InputText placeholder="change" onChange={onChangeMock} />);
  fireEvent(
    screen.getByPlaceholderText('change'),
    'onChangeText',
    'text change',
  );
  expect(onChangeMock).toHaveBeenCalledWith('text change');
});

test('focus text input event', () => {
  const onFocusMock = jest.fn();
  render(<InputText placeholder="focus placeholder" onFocus={onFocusMock} />);
  fireEvent(
    screen.getByPlaceholderText('focus placeholder'),
    'focus',
    'focus event',
  );
  expect(onFocusMock).toHaveBeenCalledWith('focus event');
});

test('Event blur input', () => {
  const onBlurMock = jest.fn();
  render(<InputText placeholder="placeholder" onBlur={onBlurMock} />);
  fireEvent(screen.getByPlaceholderText('placeholder'), 'blur', 'blur event');
  expect(onBlurMock).toHaveBeenCalledWith('blur event');
});
