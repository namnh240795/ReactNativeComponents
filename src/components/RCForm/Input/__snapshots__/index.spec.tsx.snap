// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render disable input 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "fontFamily": "Poppins-Regular",
      "padding": 10,
    }
  }
>
  <View
    pointerEvents="none"
    style={
      Object {
        "backgroundColor": "#F8F9FA",
        "borderRadius": 10,
        "display": "flex",
        "flexDirection": "row",
        "marginBottom": 4,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "padding": 16,
        }
      }
    >
      <TextInput
        editable={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        secureTextEntry={false}
        style={
          Object {
            "fontFamily": "Poppins-Light",
            "fontSize": 16,
            "lineHeight": 24,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`Render empty input 1`] = `
<View
  style={Object {}}
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "backgroundColor": "#F8F9FA",
        "borderRadius": 10,
        "display": "flex",
        "flexDirection": "row",
        "marginBottom": 4,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "padding": 16,
        }
      }
    >
      <TextInput
        editable={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        secureTextEntry={false}
        style={
          Object {
            "fontFamily": "Poppins-Light",
            "fontSize": 16,
            "lineHeight": 24,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`Render input with prefix 1`] = `
<View
  style={Object {}}
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "backgroundColor": "#F8F9FA",
        "borderRadius": 10,
        "display": "flex",
        "flexDirection": "row",
        "marginBottom": 4,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "padding": 16,
        }
      }
    >
      <TextInput
        editable={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        secureTextEntry={false}
        style={
          Object {
            "fontFamily": "Poppins-Light",
            "fontSize": 16,
            "lineHeight": 24,
          }
        }
      />
    </View>
    paste
  </View>
</View>
`;

exports[`Render input with suffix 1`] = `
<View
  style={Object {}}
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "backgroundColor": "#F8F9FA",
        "borderRadius": 10,
        "display": "flex",
        "flexDirection": "row",
        "marginBottom": 4,
      }
    }
  >
    suffix
    <View
      style={
        Object {
          "flex": 1,
          "padding": 16,
        }
      }
    >
      <TextInput
        editable={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        secureTextEntry={false}
        style={
          Object {
            "fontFamily": "Poppins-Light",
            "fontSize": 16,
            "lineHeight": 24,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`Render margin all input 1`] = `
<View
  style={
    Object {
      "marginBottom": 8,
      "marginLeft": 8,
      "marginRight": 8,
      "marginTop": 8,
    }
  }
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "backgroundColor": "#F8F9FA",
        "borderRadius": 10,
        "display": "flex",
        "flexDirection": "row",
        "marginBottom": 4,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "padding": 16,
        }
      }
    >
      <TextInput
        editable={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        secureTextEntry={false}
        style={
          Object {
            "fontFamily": "Poppins-Light",
            "fontSize": 16,
            "lineHeight": 24,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`Render margin hv input 1`] = `
<View
  style={
    Object {
      "marginBottom": 0,
      "marginLeft": 2,
      "marginRight": 2,
      "marginTop": 0,
    }
  }
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "backgroundColor": "#F8F9FA",
        "borderRadius": 10,
        "display": "flex",
        "flexDirection": "row",
        "marginBottom": 4,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "padding": 16,
        }
      }
    >
      <TextInput
        editable={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        secureTextEntry={false}
        style={
          Object {
            "fontFamily": "Poppins-Light",
            "fontSize": 16,
            "lineHeight": 24,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`Render margin width array input 1`] = `
<View
  style={
    Object {
      "marginBottom": 0,
      "marginLeft": 4,
      "marginRight": 2,
      "marginTop": 0,
    }
  }
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "backgroundColor": "#F8F9FA",
        "borderRadius": 10,
        "display": "flex",
        "flexDirection": "row",
        "marginBottom": 4,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "padding": 16,
        }
      }
    >
      <TextInput
        editable={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        secureTextEntry={false}
        style={
          Object {
            "fontFamily": "Poppins-Light",
            "fontSize": 16,
            "lineHeight": 24,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`Render padding all input 1`] = `
<View
  style={
    Object {
      "paddingBottom": 8,
      "paddingLeft": 8,
      "paddingRight": 8,
      "paddingTop": 8,
    }
  }
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "backgroundColor": "#F8F9FA",
        "borderRadius": 10,
        "display": "flex",
        "flexDirection": "row",
        "marginBottom": 4,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "padding": 16,
        }
      }
    >
      <TextInput
        editable={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        secureTextEntry={false}
        style={
          Object {
            "fontFamily": "Poppins-Light",
            "fontSize": 16,
            "lineHeight": 24,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`Render padding hv input 1`] = `
<View
  style={
    Object {
      "paddingBottom": 0,
      "paddingLeft": 2,
      "paddingRight": 2,
      "paddingTop": 0,
    }
  }
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "backgroundColor": "#F8F9FA",
        "borderRadius": 10,
        "display": "flex",
        "flexDirection": "row",
        "marginBottom": 4,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "padding": 16,
        }
      }
    >
      <TextInput
        editable={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        secureTextEntry={false}
        style={
          Object {
            "fontFamily": "Poppins-Light",
            "fontSize": 16,
            "lineHeight": 24,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`Render padding width array input 1`] = `
<View
  style={
    Object {
      "paddingBottom": 0,
      "paddingLeft": 4,
      "paddingRight": 2,
      "paddingTop": 0,
    }
  }
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "backgroundColor": "#F8F9FA",
        "borderRadius": 10,
        "display": "flex",
        "flexDirection": "row",
        "marginBottom": 4,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "padding": 16,
        }
      }
    >
      <TextInput
        editable={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        secureTextEntry={false}
        style={
          Object {
            "fontFamily": "Poppins-Light",
            "fontSize": 16,
            "lineHeight": 24,
          }
        }
      />
    </View>
  </View>
</View>
`;
